### ISSUE TITLE
Training file issue

### ISSUE THREAD

**Issue by rahulhanotDTU** [kkmuggle/tppgtoapg#1]
Thanks for this wonderful repo ,I have a doubt regarding the training file that in this repo model in intializing every  time but it is not saved anywhere can you help me in that  thanks in advance
    # Initialize the model, loss function, and optimizer for each iteration
    model = U2NET()
    he_initialization(model)
    model = model.to(device)
    optimizer = optim.Adam(model.parameters(), lr=0.00005)

    epoch_train_losses = []
    epoch_pearson_coefficients = []

    predictions = None  # Initialize the 'predictions' variable
    ground_truths = None  # Initialize the 'ground_truths' variable

    for epoch in tqdm(range(200), desc=f"Training {train_file}"):
        train_loss = train_model(model, train_dataloader, optimizer, device)
        epoch_train_losses.append(train_loss)

        # Record Pearson coefficient for test dataset
        predictions, ground_truths = predict_model(model, test_dataloader, device)
        coeff = calculate_pearson_coefficient(predictions, ground_truths)
        epoch_pearson_coefficients.append(coeff)

    train_losses.append(epoch_train_losses)
    pearson_coefficients.append(epoch_pearson_coefficients)

    if predictions is not None and ground_truths is not None:  # Save predictions and ground_truths only if they are defined
        with open(os.path.join(predict_data_folder, f"predictions_and_ground_truth_{test_file}"), "wb") as f:
            pickle.dump((predictions, ground_truths), f)

    # Plot the training loss
    plt.figure()
    plt.plot(epoch_train_losses)
    plt.xlabel('Epoch')
    plt.ylabel('Training Loss')
    plt.title(f'Training Loss - {train_file}')
    plt.savefig(f'trainloss_u2net/training_loss_{train_file}.png')  # Save the plot as an image
    plt.show()  # Display the plot

    # Plot the Pearson coefficients
    plt.figure()
    plt.plot(epoch_pearson_coefficients)
    plt.xlabel('Epoch')
    plt.ylabel('Pearson Coefficient')
    plt.title(f'Pearson Coefficient for Test Dataset - {test_file}')
    plt.savefig(f'coeff_u2net/pearson_coefficients_{test_file}.png')  # Save the plot as an image
    plt.show()  # Display the plot
